crates/bevy_asset/src/asset_server.rs

```diff
@@ -295,7 +306,9 @@ impl AssetServer {
         self.server
             .task_pool
             .spawn(async move {
-                server.load_async(owned_path, force).await.unwrap();
+                if let Err(err) = server.load_async(owned_path, force).await {
+                    warn!("Asset failed to load: {:?}", err);
```

Comment:
I think its probably better to rely on the error's "display" implementation here:

```rust
warn!("{}", err);
```

But that requires some changes to the error in question:
* rename PathLoader to AssetIo (this was already out of sync. PathLoader was an old name for the AssetIo abstraction)
* add the actual AssetIo error to the message

```rust
    #[error("AssetIo encountered an error: {0:?}")]
    AssetIoError(#[from] AssetIoError),
```

Reply:
And if you're feeling generous, you could update the other error messages in that enum to also include their contents (where relevant).

Reply:
Arg actually maybe including the "Asset failed to load" message is a good idea because the error unfortunately doesn't encapsulate that information.

The current Error impl is shared everywhere, which is nice for code size / error handling ergonomics, but makes cases like this nastier.

Feel free to leave this as-is for now (although renaming PathLoader to AssetIo would be nice :smile: )

Reply:
Good suggestions, I updated the error text of `AssetServerError` and `AssetIoError` to include the contents and use `Display` where appropriate. Now, I get a warnings that look like:

```
Jan 23 11:24:25.658  WARN bevy_asset::asset_server: encountered an error while reading an asset: path not found: /Users/will/Code/crateton/assets/models/monkey/config.json
```